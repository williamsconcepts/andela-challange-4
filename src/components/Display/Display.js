import React from 'react';

import classes from './Display.module.css';

const display = (props) => {
  const { results, format } = props;
  const { impact, severeImpact } = results;
  return (
    <div className={classes.Display}>
      <h2 className="tc f2 mb4">Estimated Results</h2>
      <div className={classes.Results}>
        <div className={classes.Impact}>
          <h3>Impact</h3>
          <div>
            <span className={classes.Title}>Currently Infected: </span>
            <span>{format(impact.currentlyInfected) || format(6740)}</span>
          </div>
          <div>
            <span className={classes.Title}>Infections By Requested Time: </span>
            <span>{format(impact.infectionsByRequestedTime) || format(3533701120)}</span>
          </div>
          <div>
            <span className={classes.Title}>Severe Cases By Requested Time: </span>
            <span>{format(impact.severeCasesByRequestedTime) || format(530055168)}</span>
          </div>
          <div>
            <span className={classes.Title}>Hospital Beds By Requested Time: </span>
            <span>{format(impact.hospitalBedsByRequestedTime) || format(-529571953)}</span>
          </div>
          <div>
            <span className={classes.Title}>Cases For ICU By Requested Time: </span>
            <span>{format(impact.casesForICUByRequestedTime) || format(176685056)}</span>
          </div>
          <div>
            <span className={classes.Title}>Cases For Ventilators By Requested Time: </span>
            <span>{format(impact.casesForVentilatorsByRequestedTime) || format(70674022)}</span>
          </div>
          <div>
            <span className={classes.Title}>Dollars InFlight: </span>
            <span>{format(impact.dollarsInFlight) || format(727589060608)}</span>
          </div>
        </div>

        <div className={classes.Impact}>
          <h3>Severe Impact</h3>
          <div>
            <span className={classes.Title}>Currently Infected: </span>
            <span>{format(severeImpact.currentlyInfected) || format(33700)}</span>
          </div>
          <div>
            <span className={classes.Title}>Infections By Requested Time: </span>
            <span>{format(severeImpact.infectionsByRequestedTime) || format(17668505600)}</span>
          </div>
          <div>
            <span className={classes.Title}>Severe Cases By Requested Time: </span>
            <span>{format(severeImpact.severeCasesByRequestedTime) || format(2650275840)}</span>
          </div>
          <div>
            <span className={classes.Title}>Hospital Beds By Requested Time: </span>
            <span>
              {format(severeImpact.hospitalBedsByRequestedTime) || format(-2649792625)}
            </span>
          </div>
          <div>
            <span className={classes.Title}>Cases For ICU By Requested Time: </span>
            <span>{format(severeImpact.casesForICUByRequestedTime) || format(883425280)}</span>
          </div>
          <div>
            <span className={classes.Title}>Cases For Ventilators By Requested Time: </span>
            <span>
              {format(severeImpact.casesForVentilatorsByRequestedTime) || format(353370112)}
            </span>
          </div>
          <div>
            <span className={classes.Title}>Dollars InFlight: </span>
            <span>{format(severeImpact.dollarsInFlight) || format(3637945303040)}</span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default display;
