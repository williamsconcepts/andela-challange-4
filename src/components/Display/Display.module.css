.Display {
  display: flex;
  flex-direction: column;
  width: 85%;
  margin: 2rem auto;
  padding: 3rem;
  background-color: white;
  border-radius: 1rem;
}

.Display .Results {
  display: flex;
  justify-content: space-between;
  font-size: 1.6rem;
}

.Display .Results .Impact {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.Display .Results .Impact div {
  margin: .5rem 0;
}

.Display .Impact .Title {
  font-size: 1.3rem;
}