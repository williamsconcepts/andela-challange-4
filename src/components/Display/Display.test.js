import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import Display from './Display';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('Testing Display Component', () => {
  const results = {
    data: {},
    impact: {},
    severeImpact: {}
  };
  const formatNumber = (x) => (x ? x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',') : x);
  test('expect to render display correctly', () => {
    expect.assertions(1);
    expect(shallow(<Display results={results} format={formatNumber} />)).toMatchSnapshot();
  });
});
