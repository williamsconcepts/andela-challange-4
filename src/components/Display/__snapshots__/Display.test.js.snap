// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Display Component expect to render display correctly 1`] = `
<div
  className="Display"
>
  <h2
    className="tc f2 mb4"
  >
    Estimated Results
  </h2>
  <div
    className="Results"
  >
    <div
      className="Impact"
    >
      <h3>
        Impact
      </h3>
      <div>
        <span
          className="Title"
        >
          Currently Infected: 
        </span>
        <span>
          6,740
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Infections By Requested Time: 
        </span>
        <span>
          3,533,701,120
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Severe Cases By Requested Time: 
        </span>
        <span>
          530,055,168
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Hospital Beds By Requested Time: 
        </span>
        <span>
          -529,571,953
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Cases For ICU By Requested Time: 
        </span>
        <span>
          176,685,056
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Cases For Ventilators By Requested Time: 
        </span>
        <span>
          70,674,022
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Dollars InFlight: 
        </span>
        <span>
          727,589,060,608
        </span>
      </div>
    </div>
    <div
      className="Impact"
    >
      <h3>
        Severe Impact
      </h3>
      <div>
        <span
          className="Title"
        >
          Currently Infected: 
        </span>
        <span>
          33,700
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Infections By Requested Time: 
        </span>
        <span>
          17,668,505,600
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Severe Cases By Requested Time: 
        </span>
        <span>
          2,650,275,840
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Hospital Beds By Requested Time: 
        </span>
        <span>
          -2,649,792,625
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Cases For ICU By Requested Time: 
        </span>
        <span>
          883,425,280
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Cases For Ventilators By Requested Time: 
        </span>
        <span>
          353,370,112
        </span>
      </div>
      <div>
        <span
          className="Title"
        >
          Dollars InFlight: 
        </span>
        <span>
          3,637,945,303,040
        </span>
      </div>
    </div>
  </div>
</div>
`;
